File: backend/app/models/transcript.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from .meeting import Language


class TranscriptSegment(BaseModel):
    start: float
    end: float
    text: str


class Transcript(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    meeting_id: UUID
    full_text: str
    segments: list[TranscriptSegment]
    language: Language | None = None
    created_at: datetime

File: frontend/src/types/transcript.ts
import type { Language } from './meeting';

export interface TranscriptSegment {
  start: number;
  end: number;
  text: string;
}

export interface Transcript {
  id: string;
  meeting_id: string;
  full_text: string;
  segments: TranscriptSegment[];
  language: Language | null;
  created_at: string;
}
